lib/msal-node/src/error/NodeAuthError.ts

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { AuthError } from "@azure/msal-common";
+
+/**
+ * NodeAuthErrorMessage class containing string constants used by error codes and messages.
+ */
+export const NodeAuthErrorMessage = {
+    addressWrongType: {
+        code: "address_wrong_type",
+        desc: "Server address is not type string. This is unexpected."
+    },
+    hostHeaderMissing: {
+        code: "host_header_missing",
+        desc: "Host header is missing in the request. This is unexpected."
+    }
+};
+
+export class NodeAuthError extends AuthError {
+    constructor(errorCode: string, errorMessage?: string) {
+        super(errorCode, errorMessage);
+        this.name = "NodeAuthError";
+
+        Object.setPrototypeOf(this, NodeAuthError.prototype);
```

Comment:
nit: This is pretty tricky, I'd add a comment about prototype chain here...

Reply:
Thanks for pointing this out...lazy copy/paste from some of our other error classes, don't think it's needed here so I've removed it.